src/helpers.php

```diff
@@ -246,3 +246,33 @@ function is_countable($obj)
         return is_array($obj) || $obj instanceof Countable;
     }
 }
+
+if (! function_exists('oldValueDefaultOrFallback')) {
+    /**
+     * This function allows us to setup a default value in case there is an old value, but is forcelly null
+     * by Laravel middleware ConvertEmptyStringsToNull.
+     *
+     * @param  string  $field_name
+     * @param  mixed  $fallback
+     * @return mixed
+     */
+    function oldValueDefaultOrFallback($field, $fallback, $field_name = null, $default = null, $value = null)
```

Comment:
I've just tested this and it looks like this solves one problem when validation fails:
- empty input is populated with _previous value_;

But it may add a different problem:
- empty input is populated with the _default value_;

----

This isn't immediately obvious, since the `default` is usually "empty string", but if you change the `default` to something else... it will be. For example:
- in Monsters, I set the `default` for the `text` field to `nooooo`;
- I go change things in the form, clear the `text` input;
- validation fails;
- when I come back, my input is NOT empty as I submitted it, instead it's `noooo`;

-----

What do I think about this?

- On one hand... it's better 😅 
    - it doesn't look like Backpack "forgot" to change that field;
    - the default value is a better unexpected value than the previous value;
    - one could argue that this is actually useful, Backpack is basically suggesting to the user... "_Oh you didn't really mean to send this empty, did you? What you wanted I think was the default value._"
- On the other hand... is this expected? I would _expect_ that field to be exactly what I submitted it, _not_ the default value... Wouldn't you @pxpm ? What do you think about this?

Reply:
No no no no!

That is a bug and I think I introduced it in the last commit. You expect the input to get the value you submitted. 

Let me have a look I think I know what is the problem.

Reply:
@tabacitu it's fixed!

Reply:
This last push broke my `relationship_select` on Edit:

![Screenshot 2021-12-23 at 18 50 28](https://user-images.githubusercontent.com/1032474/147270120-4b387f7b-620e-41be-9150-f04061dfc433.png)